frontend/src/shared/components/TopAppBar.tsx

```diff
@@ -0,0 +1,67 @@
+import { useRouter } from '@tanstack/react-router';
+import { IcArrowLeft, IcCloseM } from '@icons';
+
+export interface TopAppBarProps {
+  title?: string;
+  showBack?: boolean;
+  showClose?: boolean;
+  background?: boolean;
+  onClose?: () => void;
+  className?: string;
+}
+
+const TopAppBar = ({
+  title,
+  showBack = false,
+  showClose = false,
+  background = true,
+  onClose,
+  className = '',
+}: TopAppBarProps) => {
+  const router = useRouter();
+  const handleBack = () => {
+    router.history.back();
+  };
+
+  const backgroundClass = background
+    ? 'bg-[var(--color-background-default-white)]'
+    : 'bg-transparent';
+
+  return (
+    <header className={`flex-between h-[4.8rem] w-full px-[2rem] ${backgroundClass} ${className}`}>
+      <div className='flex h-[4rem] w-[4rem] items-center justify-start'>
+        {showBack && (
+          <button
+            type='button'
+            onClick={handleBack}
+            className='flex-center text-text-neutral-primary'
+            aria-label='뒤로 가기'>
+            <IcArrowLeft className='h-[2.4rem] w-[2.4rem]' />
+          </button>
+        )}
+      </div>
+
+      <div className='flex-center flex-1'>
+        {title && (
+          <h1 className='subtitle-18-bold text-text-neutral-primary cursor-default text-center'>
+            {title}
+          </h1>
+        )}
+      </div>
+
+      <div className='flex h-[4rem] w-[4rem] items-center justify-end'>
```

Comment:
```suggestion
      <div className='flex-end h-[4rem] w-[4rem]'>
```

이 친구들도 flex-utility ㅎㅎ,,